From 308df829717768c7448c3f81ed0a41b057a2451a Mon Sep 17 00:00:00 2001
From: Chad Dupuis <cdupuis@redhat.com>
Date: Mon, 18 Aug 2014 19:56:07 -0400
Subject: [scsi] qla2xxx: Only complete dcbx_comp and lb_portup_comp for virtual port index 0

Message-id: <1408391817-6389-20-git-send-email-cdupuis@redhat.com>
Patchwork-id: 87911
O-Subject: [PATCH RHEL 7.1 e-stor 19/69] qla2xxx: Only complete dcbx_comp and lb_portup_comp for virtual port index 0.
Bugzilla: 1089346
RH-Acked-by: Rob Evers <revers@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Chad Dupuis <chad.dupuis@qlogic.com>

scsi commit id 9aaf2cea4e63ed5ccec728f550e87bab61a571e4

Signed-off-by: Chad Dupuis <chad.dupuis@qlogic.com>
Signed-off-by: Saurav Kashyap <saurav.kashyap@qlogic.com>
Signed-off-by: James Bottomley <JBottomley@Parallels.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/qla2xxx/qla_isr.c b/drivers/scsi/qla2xxx/qla_isr.c
index 41de008..a802626 100644
--- a/drivers/scsi/qla2xxx/qla_isr.c
+++ b/drivers/scsi/qla2xxx/qla_isr.c
@@ -294,7 +294,7 @@ qla81xx_idc_event(scsi_qla_host_t *vha, uint16_t aen, uint16_t descr)
  case MBA_IDC_COMPLETE:
   if (mb[1] >> 15) {
    vha->hw->flags.idc_compl_status = 1;
-   if (vha->hw->notify_dcbx_comp)
+   if (vha->hw->notify_dcbx_comp && !vha->vp_idx)
     complete(&vha->hw->dcbx_comp);
   }
   break;
@@ -765,7 +765,7 @@ skip_rio:
    ql_dbg(ql_dbg_async, vha, 0x500d,
        "DCBX Completed -- %04x %04x %04x.\n",
        mb[1], mb[2], mb[3]);
-   if (ha->notify_dcbx_comp)
+   if (ha->notify_dcbx_comp && !vha->vp_idx)
     complete(&ha->dcbx_comp);
 
   } else
@@ -1039,7 +1039,7 @@ skip_rio:
    }
   }
  case MBA_IDC_COMPLETE:
-  if (ha->notify_lb_portup_comp)
+  if (ha->notify_lb_portup_comp && !vha->vp_idx)
    complete(&ha->lb_portup_comp);
   /* Fallthru */
  case MBA_IDC_TIME_EXT:
-- 
1.7.1